### PR TITLE
feat: #tno-1817 - allow section content grouping

### PR DESCRIPTION
**Pull request by kylgarmor**
- updated reportsection component to add Group By section
- added a few helper methods to use within the Razor template for grouping

### CODE REVIEW

**Comment by kylgarmor** [comment]
- update to template to occur independently post merge

File: app/editor/src/features/admin/reports/ReportSection.tsx [hunk 1]
```diff
@@ -261,6 +261,22 @@ export const ReportSection: React.FC<IReportSectionProps> = ({ className, index
                 {folder?.description && <p>{folder?.description}</p>}
               </Col>
             </Row>
+            <Row>
+              <Col flex="1" className="description">
+                <FormikSelect
+                  label="Group Section Content By"
+                  name={`settings.${index}.settings.groupBy`}
```
Comment (Fosol) [inline app/editor/src/features/admin/reports/ReportSection.tsx:268]: This doesn't look correct `settings.${index}.settings.groupBy`.  Did you mean `sections...`?

**Review by Fosol — COMMENTED** [review]
Do you need to make any changes to the Custom Report Template?

File: app/editor/src/features/admin/reports/ReportSection.tsx (hunk 1 above)
Comment (kylgarmor) [reply app/editor/src/features/admin/reports/ReportSection.tsx:268]: you are correct

**Review by Fosol — APPROVED** [review]

**Pull request merged**